### PR TITLE
fix: throws Element's lifecycle rejections in the React render cycle

### PR DESCRIPTION
**Pull request by djejaquino**
## Description

Throws Element's lifecycle rejections `mount` and `update` at React render cycle

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: src/elements/useElement.ts [hunk 1]
```diff
@@ -58,7 +58,11 @@ const useElement = <
     if (bt && !element) {
       const newElement = bt.createElement(type as never, options as never);
 
-      newElement.mount(`#${id}`);
+      newElement.mount(`#${id}`).catch((mountError) => {
+        setElement(() => {
+          throw mountError;
+        });
```
[the remark this answers is not in the document]
Comment (djejaquino) [reply src/elements/useElement.ts:64]: 🧙‍♂️

File: test/elements/useElement.test.ts [hunk 2]
```diff
@@ -18,9 +18,9 @@ describe('useElement', () => {
   let update: jest.Mock;
 
   beforeEach(() => {
-    mount = jest.fn();
+    mount = jest.fn(() => Promise.resolve());
```
Comment (brigonzalez) [inline test/elements/useElement.test.ts:21]:
```suggestion
    mount = jest.fn().mockResolvedValue();
```
nbd if you don't take these

File: test/elements/useElement.test.ts [hunk 3]
```diff
@@ -18,9 +18,9 @@ describe('useElement', () => {
...
 
-    update = jest.fn();
+    update = jest.fn(() => Promise.resolve());
```
Comment (brigonzalez) [inline test/elements/useElement.test.ts:23]:
```suggestion
    update = jest.fn().mockResolvedValue();
```

**Review by brigonzalez — DISMISSED** [review]

File: test/elements/useElement.test.ts (hunk 2 above)
Comment (djejaquino) [reply test/elements/useElement.test.ts]: I was back and forth on this because it requires a value.

Comment (djejaquino) [reply test/elements/useElement.test.ts]: Accepted tho

**Review by brigonzalez — APPROVED** [review]

**Pull request merged**